### DISCUSSION TITLE
Infrastructure to maintain a module version and interactions among modules

### DISCUSSION THREAD

**Discussion by danolivo** [pgedc/pgedc#45] [Ideas]
Having lots of Postgres extensions/modules we should end up with more mature managing infrastructure. For now, I see necessity to add into the module information about its version. What's more, I think in the near future we need to invent some callback to register on a external library load event. It will help to identify and resolve known conflicts among modules.
Right now I have next real-world cases causing problems:

1. Fresh minor version of database working on previous minor version of binaries.
2. Module supports different versions of an extension (for example, 2.3 - 2.7). User have version 2.5 installed. But what exactly is a version of the module code?
3. Module conflicts with specific version of another module.

Do you know (or imagine) any extra cases? Write it down in comments, please.

**Comment by danolivo** [comment]
@yrashk ,
I have written the code for the Postgres core - see the [branch](https://github.com/danolivo/pgdev/tree/module-info) or 
[v1-0001-Introduce-MODULE_INFO-macro.patch](https://github.com/user-attachments/files/17824309/v1-0001-Introduce-MODULE_INFO-macro.patch) for details.
 It would be nice to arrange the Extensions Developers Ccommunity's standpoint of this feature before pushing any proposal to the pgsql-hackers mailing list.

**Reply by yrashk** [reply]
@danolivo will take a good look at it shortly, thank you!
